### PR TITLE
Remove dead code around x86 delegate interop.

### PR DESCRIPTION
**Pull request by jkoritzinsky**
We no longer use any intercept stubs on x86, so we can remove the infra that supported using them.

### CODE REVIEW

File: src/coreclr/vm/syncblk.h [hunk 1]
```diff
@@ -729,50 +729,29 @@ class InteropSyncBlockInfo
...
     {
         LIMITED_METHOD_CONTRACT;
-        return (((UINT_PTR)m_pUMEntryThunkOrInterceptStub & 1) ? NULL : m_pUMEntryThunkOrInterceptStub);
-    }
-
-    Stub* GetInterceptStub()
-    {
-        LIMITED_METHOD_CONTRACT;
-        return (((UINT_PTR)m_pUMEntryThunkOrInterceptStub & 1) ? (Stub *)((UINT_PTR)m_pUMEntryThunkOrInterceptStub & ~1) : NULL);
+        return (((UINT_PTR)m_pUMEntryThunk & 1) ? NULL : m_pUMEntryThunk);
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/vm/syncblk.h:748]: Do we still set the low order bit during delete?

Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/syncblk.h:748]: The `SetInterceptStub` API did above, but about to the thunk itself?

**Comment by jkotas** [comment]
LGTM module Aaron's comment.

File: src/coreclr/vm/syncblk.h (hunk 1 above)
Comment (jkoritzinsky) [reply src/coreclr/vm/syncblk.h:748]: We don't. I'll clean this up.

**Review by elinor-fung — APPROVED** [review]

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Pull request merged**